tests/test_functional.py

```diff
@@ -61,16 +60,17 @@ def test_QueryExecutionerWorkflow(query):
     # whether this metadata is actually present. And that it only contains
     # meta data related to his query
 
-    assert type(res_query1.meta_data) == dict, "meta data not a dict"
```

Comment:
I think the meta data checking should be part of a separate query execution test. The high level functional test should just check whether the end result used meta data correct according to the desired post-processing. (Which id does not do yet at all).